filters/IterativeClosestPoint.cpp

```diff
@@ -165,7 +165,11 @@ PointViewPtr IterativeClosestPoint::icp(PointViewPtr fixed,
         fixed_idx.reserve(tempMovingTransformed->size());
         moving_idx.reserve(tempMovingTransformed->size());
         double mse(0.0);
-        double sqr_maxdist = m_maxdist * m_maxdist;
+        double sqr_maxdist;
+        if (m_maxdistArg->set())
+            sqr_maxdist = m_maxdist * m_maxdist;
+        else
+            sqr_maxdist = std::numeric_limits<double>::max();
```

Comment:
This needs to be wrapped to avoid macro expansion:

`(std::numeric_limits<double>::max)()`